### PR TITLE
travis: add linux-ppc64le and osx

### PR DESCRIPTION
**Pull request by chavafg**
Run statick checks on linux-ppc64le and osx.

Fixes: #691.

Signed-off-by: Salvador Fuentes <salvador.fuentes@intel.com>

### CODE REVIEW

**Comment by chavafg** [comment]
getting this error:
```
.ci/static-checks.sh: line 24: typeset: -A: invalid option
```
but that one should not be a problem if installing latest available `bash`, right? 
any idea @jodh-intel @raravena80?

**Comment by raravena80** [comment]
@chavafg right. But brew installs it under `/usr/local/bin/bash` and not `/usr/bin/bash` (which is the system bash)

**Comment by marcov** [comment]
@raravena80 right, so @chavafg should probably add something like:
```
before_script:
  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then export PATH=/usr/local/bin:$PATH; fi
```

**Comment by chavafg** [comment]
ohh right, thank you both, let me add that line.

**Comment by marcov** [comment]
Also you need to change the shebang in `.ci/static_checks.sh` with `#!/usr/bin/env bash`

**Comment by jodh-intel** [comment]
Thanks @chavafg!

lgtm

/cc @raravena80.

**Comment by chavafg** [comment]
Seems like `SIGSTKFLT ` and `SIGPWR` are not available in Darwin, any idea of how can I skip those if running on Darwin, but use them if running on Linux?

```
integration/docker/kill_test.go:135:14:warning: SIGSTKFLT not declared by package syscall (staticcheck)
integration/docker/kill_test.go:135:14:warning: unused variable or constant SIGSTKFLT not declared by package syscall (varcheck)
integration/docker/kill_test.go:149:14:warning: unused variable or constant SIGPWR not declared by package syscall (varcheck)
integration/docker/kill_test.go:135:14:warning: unused struct field SIGSTKFLT not declared by package syscall (structcheck)
integration/docker/kill_test.go:149:14:warning: unused struct field SIGPWR not declared by package syscall (structcheck)
integration/docker/kill_test.go:135:14:warning: SIGSTKFLT not declared by package syscall (unused)
```

**Comment by caoruidong** [comment]
Split them into two files with +build tags?

**Comment by marcov** [comment]
@chavafg you can safely remove SIGSTKFLT (it's unused on Linux and not existing on darwin).

About SIGPWR, is not a POSIX signal and maybe that can be removed too. From signal(7):
```
       SIGPWR (which is not specified in POSIX.1-2001) is typically ignored by default on those other  UNIX  systems  
       where it appears
```

File: cmd/log-parser/Makefile [hunk 1]
```diff
@@ -22,8 +24,14 @@ check:
 $(TARGET): $(SOURCES) check
 	go build -o "$(TARGET)" -ldflags "-X main.name=${TARGET} -X main.commit=${COMMIT} -X main.version=${VERSION}" .
 
+INSTALL_FLAGS :=
+
+ifeq ($(detected_OS),Linux)
+	INSTALL_FLAGS := -D
```
Comment (marcov) [inline cmd/log-parser/Makefile]: To keep things cleaner you can avoid checking OS version here and just run:

```Makefile
install: $(TARGET)
	install -d $(shell dirname $(DESTTARGET))
	install $(TARGET) $(DESTTARGET)
```

**Comment by jodh-intel** [comment]
Agreed - we've already removed `SIGSTKFLT` from the shim and proxy:

- https://github.com/kata-containers/shim/pull/87
- https://github.com/kata-containers/proxy/pull/82

We still have `SIGPWR` in the runtime and the default behaviour appears to be "terminate". To handle that for Darwin, you could adopt the approach we're using in the runtime which:

- Creates architecture-specific `.go` files.
- Creates variables and functions prefixed with `arch*`. Those variables and functions then either provide a custom (architecture-specific) implementation, *or* they are set to a function called `generic*` to provide the default behaviour.

Specifically, I think you could:

- Move the signal and bool values in the `withSignal()` calls into a `var genericSignalMap map[syscall.Signal]bool`, but don't add `SIGPWR` to the map.
- Create `integration/docker/kill_linux_test.go` which will:
  - Create a `var archSignalMap = genericSignalMap`.
  - Add a `syscall.SIGPWR` entry to the map (maybe in a `func init()` function).
  - Call the common `Describe()` function in  `integration/docker/kill_test.go`.
- Create `integration/docker/kill_darwin_test.go` which will:
  - Create a `var archSignalMap = genericSignalMap`.
  - Call the common `Describe()` function in  `integration/docker/kill_test.go`.
- Rework the `Describe()` function to iterate the `archSignalMap`, calling `WithSignal()` on the values.

**Comment by jodh-intel** [comment]
Thanks @chavafg!

lgtm

**Comment by GabyCT** [comment]
lgtm

**Pull request merged**